include/zephyr/toolchain.h

```diff
@@ -318,6 +318,18 @@
 #define TOOLCHAIN_ENABLE_IAR_WARNING(warning)
 #endif
 
+/**
+ * @def TOOLCHAIN_DEPRECATED
+ * @brief Marks a function or symbol as deprecated as of @p removal_version.
+ *
+ * @param removal_version Version of planned removal
+ */
+#if defined(__GNUC__) || defined(__llvm__)
+#define TOOLCHAIN_DEPRECATED(removal_version) __deprecated_version(removal_version)
+#else
+#define TOOLCHAIN_DEPRECATED(removal_version)
+#endif
```

Comment:
This is pretending to be one of the toolchain warning control macros when it is not -- `TOOLCHAIN_DEPRECATED` is unnecessary and should be removed.

Only `__deprecated_version` should be used in code and that should be made available in every toolchain supported.